src/v/cluster/partition_recovery_manager.cc

```diff
@@ -317,12 +319,14 @@ ss::future<log_recovery_result> partition_downloader::download_log() {
       retention);
     auto mat = co_await find_recovery_material();
     if (cst_log.is_enabled(ss::log_level::debug)) {
-        std::stringstream ostr;
-        mat.partition_manifest.serialize_json(ostr);
-        vlog(
-          _ctxlog.debug,
-          "Partition manifest used for recovery: {}",
-          ostr.str());
+        iobuf buf;
+        auto ostr = make_iobuf_ref_output_stream(buf);
+        co_await mat.partition_manifest.serialize_json(ostr);
+        auto istr = make_iobuf_input_stream(std::move(buf));
+        auto truncated = co_await istr.read_exactly(
```

Comment:
what happens if you don't truncate?

arbitrary truncation looks odd especially if we ever come to rely on this information for debugging; if it is useless (and we are fine dropping it arbitrarily) then why are we printing it at all?

Reply:
> what happens if you don't truncate?

1. It's needs additional effort to log without oversized allocations (although it absolutely is possible) 
2. Potentially it also bloats the logs
My reasoning was that if something is large enough to cause an oversized allocation (128Kb) it may also significantly bloat logs as there may be thousands of partitions to recover.

As for how useful this log line is, that's I would like @Lazin's opinion, maybe he remembers.

Reply:
dump 128kb to a single line is intense. surely there is a better solution than arbitrary truncation and dumping everything.